lc0/src/neural/network_cudnn.cu

```diff
@@ -0,0 +1,836 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2018 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "network_cudnn.h"
+#include "utils/bititer.h"
+#include <functional>
+#include <cassert>
+#include <mutex>
+
+#include <cublas_v2.h>
+#include <cudnn.h>
+
+
+void cudnnError(cudnnStatus_t status, const char* file, const int& line)
+{
+    if (status != CUDNN_STATUS_SUCCESS)
+    {
+        printf("CUDNN error: %s (%s:%d)\n", cudnnGetErrorString(status), file, line);
```

Comment:
Consider throwing an exception (from utils/exception.h) instead, that way UCI loop has a chance to recover.

Reply:
done.